### PR TITLE
python3Packages.dagger-io: init at 0.6.1

### PR DESCRIPTION
**Pull request by rayanpiro**
###### Description of changes

New python module include.
Name: [dagger-io](https://dagger.io/)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (Janik-Haag) [inline pkgs/development/python-modules/dagger-io/default.nix:4]: can you please remove this, since it is not needed?

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (Janik-Haag) [inline pkgs/development/python-modules/dagger-io/default.nix:24]: Please remove the unneded line:
```suggestion
```

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (Janik-Haag) [inline pkgs/development/python-modules/dagger-io/default.nix:23]: this should be alphabetically ordered.  Except for the lib attribute that should be at the top.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (Janik-Haag) [inline pkgs/development/python-modules/dagger-io/default.nix:41]: please ad a comment describing why you are disabling the test? And adding `pythonImportsCheck` would be neat.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (Janik-Haag) [inline pkgs/development/python-modules/dagger-io/default.nix:59]: you are missing meta attributes see for example [here](https://github.com/NixOS/nixpkgs/blob/e83cd1fb00098aec91ff5af33a8513804ca95eb9/pkgs/development/python-modules/django-leaflet/default.nix#L32-L38)

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
Thanks for your contribution, feel free to ask questions and I will try answering them the best I can :D 

Also you should think about adding your self to the [maintainers-list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) and after that add your self in the package metadata.
```nix
meta.maintainers = with lib.maintainers; [ your_nick_name ];```

**Review by Janik-Haag — COMMENTED** [review]
Also please merge your two commits in to one called `python3Packages.dagger-io: init at 0.6.1` and if you add youself to the maintainers list call that commit: `maintainers: add your_nick_name`

**Comment by rayanpiro** [comment]
Ok, I'll try to change it tomorrow. Thanks for your feedback!

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/dagger-io/default.nix:40]: We can use `sourceRoot` here.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dagger-io/default.nix:44]:
```suggestion
```

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dagger-io/default.nix:58]:
```suggestion
```
windows only

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dagger-io/default.nix:38]:
```suggestion
  version = "v0.6.1";
  format = "pyproject";

  disabled = pythonOlder "3.10";

  src = fetchFromGitHub {
    owner = "dagger";
    repo = "dagger";
    rev = "$v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix]: Sorry Im kinda new on NixOS. Could you provide an example or documentation of how to use sourceRoot? I was looking for info and just got working with this.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix:53]: Without colorama build fails.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix]: Thank you so much, was fixed :+1:

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix:39]: Not building without it :cry:

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix]: Was totally a mistake, it works perfectly with tests, added.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix:52]: Added meta, just let commented the maintainer.
I dont know if I need to wait until maintainers-list is merged to add it or not.
Thanks for your time :smile:

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [reply pkgs/development/python-modules/dagger-io/default.nix]: Sure. What about the [unpackPhase section of the nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#ssec-unpack-phase)?
Examples can be found by running `grep sourceRoot pkgs/development/python-modules/*/*.nix`.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [reply pkgs/development/python-modules/dagger-io/default.nix]: He means `poetry-core` should go to `nativeBuildInputs`.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [reply pkgs/development/python-modules/dagger-io/default.nix]: We accidentally used `poetry-core` instead of `hatchling`, so `requirements.txt` may be loaded.

The upstream build-system is defined [here](https://github.com/dagger/dagger/blob/9ec0d6a9625c85655e8d82f241771470fe6998aa/sdk/python/pyproject.toml#L1-L3).

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/dagger-io/default.nix:24]:
```suggestion
  version = "0.6.1";
```
`v` is a prefix, and it should not be included in the version.

**Review by rayanpiro — COMMENTED** [review]
Thanks for the feedback, I think it will be complete with these two small changes.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix]: Ok, Ill take a look, thank you so much for your time, is really helpfull.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dagger-io/default.nix:59]:
```suggestion
    maintainers = with maintainers; [ rayanpiro ];
```
this is immediate with this PR

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dagger-io/default.nix:33]:
```suggestion
    owner = "dagger";
    repo = "dagger";
    rev = "v${version}";
    hash = "sha256-9QQ6aDCkTWNq5KOSGF6FH6UQrOYa51ctW3CMcGrCJAQ=";
```

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dagger-io/default.nix:46]: You are sure poetry-core is not only used to build the package? Then it should go to nativeBuildInputs

Comment (Janik-Haag) [reply pkgs/development/python-modules/dagger-io/default.nix:46]: It doesn't build without so I think it's fine to leave it.

**Review by Janik-Haag — COMMENTED** [review]
Please squash the commits so you only have `maintainers: rayanpiro` and `python3Packages.dagger-io: init at 0.6.1` after that it should be good to go @natsukium  what do you think?

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/dagger-io/default.nix:34]:
```suggestion
  inherit (dagger) src;
  
  sourceRoot = "source/sdk/python";
```
It seems there is already `dagger` in nixpkgs.
How about using its src?

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [reply pkgs/development/python-modules/dagger-io/default.nix:46]: `poetry-core` should go to `nativeBuildInputs` since it is a build-time only dependency.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/dagger-io/default.nix:54]:
```suggestion
    description = "Dagger is a programmable CI/CD engine that runs your pipelines in containers";
```
please remove the trailing period.
https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix:34]: I've no idea of how can I reference the source/sdk/python of another nixpkg.

The existing dagger package I think is the CLI tool. So I've assumed that I've to fetch from github the python sdk from the dagger-io repository to build the package. I just guided myself with other nixpkgs that does similar.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (rayanpiro) [reply pkgs/development/python-modules/dagger-io/default.nix:45]: I'll try it tomorrow and change it to if it works.

File: pkgs/development/python-modules/dagger-io/default.nix
Comment (natsukium) [reply pkgs/development/python-modules/dagger-io/default.nix:34]: hnswlib should be helpful.
hnswlib is a C++ header library, but also provides a python module.
https://github.com/NixOS/nixpkgs/blob/8d56c1fe5a61939290694404947fa4127e7ced2f/pkgs/development/python-modules/hnswlib/default.nix
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/hnswlib/default.nix

**Comment by Mic92** [comment]
Sorry. This change violate the trademark: https://github.com/NixOS/nixpkgs/issues/260848

**Comment by rayanpiro** [comment]
Ok, removed the fork too.

> It might make sense to explore whether the SDK redistribution violates the trademark.

It seems weird to me, however if you prohibit using the package name in a package manager you have weird trademark policies.

**Pull request closed without merging**
